### PR TITLE
[libftdi1] update to 1.5 (#15917)

### LINKED ISSUE #15917: [libftdi1] update to 1.5
Library name: libftdi1

New version number: 1.5

Release notes:

Implement tc[io]flush methods & deprecate broken purge_buffers methods
Please check your code for ftdi_usb_purge_rx_buffer(), ftdi_usb_purge_tx_buffer() and ftdi_usb_purge_buffers() and migrate to the new ftdi_tc[io]flush() methods.

Old code will continue to function, but you'll get a deprecation warning during compilation.

Add program to test buffer flush (purge) functionality
Add kernel driver auto attach/detach.
See new AUTO_DETACH_REATACH_SIO_MODULE option
Add ftdi_setflowctrl_xonxoff()
ftdi_eeprom / eeprom handling:
Unify handling of all boolean eeprom flags
Add device release number support
Add channel_a_driver support for type xxR chips
Add support for group0 drive levels on x232H chips
Fix handling of high_current_drive parameter
Fix inverted handling of VCP driver field for TYPE_R chips
New --verbose option for eeprom decode operation
Add example code for async mode
Add SPDX license identifiers to the core library & ftdi_eeprom
Various python SWIG wrapper improvements
Various cmake file improvements
Fix small bugs in error code paths

### PR DESCRIPTION
**Pull request by eloc3147**
Updates libftdi1 to version 1.5

- What does your PR fix? Fixes #15917 

- Which triplets are supported/not supported? Have you updated the CI baseline?
This shouldn't have changed with this update

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
As far as I can tell.

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libftdi1/ftdi-eeprom-fix.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeOptions.txt b/CMakeOptions.txt
+index 07b5887..43a9d3d 100755
+--- a/CMakeOptions.txt
++++ b/CMakeOptions.txt	
+@@ -3,6 +3,6 @@ option ( BUILD_TESTS "Build unit tests with Boost Unit Test framework" OFF )
+ option ( DOCUMENTATION "Generate API documentation with Doxygen" OFF )
+ option ( EXAMPLES "Build example programs" ON )
+ option ( FTDIPP "Build C++ binding library libftdi1++" OFF )
+-option ( FTDI_EEPROM "Build ftdi_eeprom" ON )
++option ( FTDI_EEPROM "Build ftdi_eeprom" OFF )
```
Comment (dan-shaw) [inline ports/libftdi1/ftdi-eeprom-fix.patch:10]: Instead of a patch, can we pass this as an option to `vcpkg_configure_cmake`?

Comment (eloc3147) [reply ports/libftdi1/ftdi-eeprom-fix.patch]: Yes, it seems so.

**Comment by eloc3147** [comment]
Current failures are referencing the "seal" port, unsure how to fix that. Any suggestions?

**Review by ras0219-msft — APPROVED** [review]
Thanks for the update!

**Pull request merged**